packages/story-editor/src/app/highlights/quickActions/useQuickActions.js

```diff
@@ -157,18 +171,43 @@ const useQuickActions = () => {
         actionLabel: __('Undo', 'web-stories'),
         dismissable: false,
         message: __('Element properties have been reset', 'web-stories'),
-        onAction: undo,
+        // Don't pass a stale version of `undo`
+        onAction: () => undoRef.current(),
       });
     },
-    [handleResetProperties, showSnackbar, undo]
+    [handleResetProperties, showSnackbar]
   );
 
+  /**
+   * Highlights a panel in the editor. Triggers a tracking event
+   * using the selected element's type.
+   *
+   * The selected element and selected element type may be overridden
+   * using the `elementParams` arguments.
+   *
+   * @param {string} highlight The panel to highlight
+   * @param {Object} elementParams
+   * @param {string} elementParams.elementId The element id that is or will be selected in the canvas.
+   * @param {string} elementParams.elementType The type of the element that is or will be selected in the canvas.
+   * @param {Event} ev The triggering event.
+   */
   const handleFocusPanel = useCallback(
-    (highlight) => (elementId) => (ev) => {
-      ev.preventDefault();
-      setHighlights({ elementId, highlight });
-    },
-    [setHighlights]
+    (highlight) =>
+      ({ elementId, elementType } = {}) =>
+      (ev) => {
+        ev.preventDefault();
+        setHighlights({
+          elementId: elementId || selectedElement?.id,
+          highlight,
+        });
+
+        trackEvent('quick_action', {
+          name: 'focus_panel',
+          element: elementType || selectedElement?.type,
+          panel: highlight,
+        });
```

Comment:
What's an example value of `highlight` here?

To me it seems a bit confusing that the quick action being clicked on is "Change color", but in telemetry it's a "focus_panel" event. Can't we track "change_color" instead?

It's especially confusing since the "Change color" and "Edit text" actions highlight the same Text panel in the Design Panel. How are these being distinguished here?

Reply:
`highlight` would be the name of the panel as defined here: https://github.com/google/web-stories-wp/blob/main/packages/story-editor/src/app/highlights/states.js#L34-L55

Yeah I agree it's a bit confusing. I can update that so that the action that's clicked is what is recorded rather than the result of the action 👍

Reply:
> Yeah I agree it's a bit confusing. I can update that so that the action that's clicked is what is recorded rather than the result of the action 👍

I think that would be more valuable for us to have in analytics. And should also help with the "Change color" and "Edit text" distinction.

Reply:
Should be fixed now 😄